compiler/bindings/python/iree/compiler/tools/import_onnx/importer_externalization_overrides.py

```diff
@@ -0,0 +1,263 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import copy
+import random
+import string
+import iree.runtime as rt
```

Comment:
I missed this during review, but this is the only place in the compiler Python bindings that depends on the runtime Python bindings. It is used for `rt.ParameterIndex` and `create_archive_file`. That makes

```bash
pip install iree-base-compiler[onnx]
```

insufficient for running `iree-import-onnx` now, as `iree-base-runtime` must also be installed

We could

* have this check the import and fail gracefully
* add a compiler API for creating parameter archives so the runtime API is not needed
* have the compiler Python package depend on the runtime Python package

Reply:
I'm working on a few improvements in <https://github.com/iree-org/iree/pull/19217>. Will make a conditional import for this in that patch.

Reply:
@ScottTodd I think it would be best to fail gracefully in case the import fails, at least for now. I am not sure about how having the compiler python package depend on the runtime package be implemented straightaway, and adding a compiler API for creating parameter archives could possibly lead to code duplication IMO.

I can create a PR for the first option right away.